_posts/2021-08-26-geometric-deep-learning/geometric_deep_learning.Rmd

```diff
@@ -0,0 +1,125 @@
+
+-   Linear group-equivariant layers. (Here group is the group of transformations whose symmetries we're interested to preserve.)
+
+-   Nonlinearities. (This really does not follow from geometric arguments, but from the observation, often stated in introductions to DL, that without nonlinearities, there is no hierarchical composition of features, since all operations can be implemented in a single matrix multiplication.)
+
+-   Local pooling layers. (These achieve the effect of coarse-graining, as enabled by the scale separation prior.)
+
+-   A group-invariant layer (global pooling). (Not every task will require such a layer to be present.)
+
+Having talked so much about the concepts, which are highly fascinating, this list may seem a bit underwhelming. That's what we've been doing anyway, right? Maybe; but once you look at a few domains and associated network architectures, the picture gets colorful again. So colorful, in fact, that we can only present a very sparse selection of highlights.
+
+## Domains, priors, architectures
+
+Given cues like "local" and "pooling", what better architecture is there to start with than CNNs, the (still) paradigmatic deep learning architecture? Probably, it's also the one a prototypic practitioner would be familiar with the most.
+
+### Images and CNNs
+
+Vanilla CNNs are easily mapped to the four types of layer that make up the blueprint. Skipping over the nonlinearities, which, in this context, are of least interest, we next have two kinds of pooling.
+
+First, a local one, corresponding to max- or average-pooling layers with small strides (2 or 3, say). This reflects the idea of successive coarse-graining, where, once we've made use of some fine-grained information, all we need to proceed is a summary.
+
+Second, a global one, used to effectively remove the spatial dimensions. In practice, this would usually be global average pooling. Here, there's an interesting detail worth mentioning. A common practice, in image classification, is to replace global pooling by a combination of flattening and one or more feedforward layers. Since with feedforward layers, position in the input matters, this will do away with translation invariance.
+
+Having covered three of the four layer types, we come to the most interesting one. In CNNs, the local, group-equivariant layers are the convolutional ones. What kinds of symmetries does convolution preserve? Think about how a kernel slides over an image, computing a dot product at every location. Say that through training, it has developed an inclination toward singling out penguin bills. It will detect, and mark, one everywhere in an image --- be it shifted left, right, top or bottom in the image. What about rotational motion, though? Since kernels move vertically and horizontally, but not in a circle, a rotated bill will be missed. Convolution is shift-equivariant, not rotation-invariant.
```

Comment:
```suggestion
Having covered three of the four layer types, we come to the most interesting one. In CNNs, the local, group-equivariant layers are the convolutional ones. What kinds of symmetries does convolution preserve? Think about how a kernel slides over an image, computing a dot product at every location. Say that, through training, it has developed an inclination toward singling out penguin bills. It will detect, and mark, one everywhere in an image --- be it shifted left, right, top or bottom in the image. What about rotational motion, though? Since kernels move vertically and horizontally, but not in a circle, a rotated bill will be missed. Convolution is shift-equivariant, not rotation-invariant.
```